From eb3bafe8a8363e962f4880789f3791590bc07353 Mon Sep 17 00:00:00 2001
From: Sean hoyt <seanhoyt963@gmail.com>
Date: Sat, 17 Feb 2018 22:37:09 +0000
Subject: [PATCH 06/10] Avoid crash when the actionbar is disabled in settings

Some devices (Wearables), disable the actionbar due to the limited
screen space on them.

Change-Id: I7282234f5baa128c438e556f80956fb7118ce8f5
---
 .../com/android/settingslib/drawer/SettingsDrawerActivity.java | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/drawer/SettingsDrawerActivity.java b/packages/SettingsLib/src/com/android/settingslib/drawer/SettingsDrawerActivity.java
index 190f5e6d140..64969aeaba1 100644
--- a/packages/SettingsLib/src/com/android/settingslib/drawer/SettingsDrawerActivity.java
+++ b/packages/SettingsLib/src/com/android/settingslib/drawer/SettingsDrawerActivity.java
@@ -75,11 +75,13 @@ public class SettingsDrawerActivity extends Activity {
         mContentHeaderContainer = (FrameLayout) findViewById(R.id.content_header_container);
 
         Toolbar toolbar = (Toolbar) findViewById(R.id.action_bar);
-        if (theme.getBoolean(android.R.styleable.Theme_windowNoTitle, false)) {
-            toolbar.setVisibility(View.GONE);
-            return;
+        if (toolbar != null) {
+            if (theme.getBoolean(android.R.styleable.Theme_windowNoTitle, false)) {
+                toolbar.setVisibility(View.GONE);
+                return;
+            }
+            setActionBar(toolbar);
         }
-        setActionBar(toolbar);
 
         if (DEBUG_TIMING) {
             Log.d(TAG, "onCreate took " + (System.currentTimeMillis() - startTime)
-- 
2.15.1

